### DISCUSSION TITLE
Add custom requested claims to user info response

### DISCUSSION THREAD

**Discussion by lucamontano98** [keycloak/keycloak#29108] [Q&A]
### Description

Good morning,
I'm trying to return specific claims which client requested but that claims are not previously defined: one of them is the fiscal number. I tried to set a mapper in the client scope view of my client, but when I send a request object which contains the claim "claims" with the claims that the resource server must return, the final response doesn't have "unknown" requested claims. I tried to define a user with these kind of claim, but the result is the same. How can I obtain all the claims that I set to users and I requested with the request object at the beginning of the OIDC authentication flow? Thanks a lot.

Luca Montano

PS: the version I'm using is the 24.0.2 and is the first I tried.

**Comment by codespearhead** [comment]
By "claims [that] are not previously defined [...]  claims that I set to users", do you mean [Unmanaged Attributes](https://www.keycloak.org/docs/latest/server_admin/#_understanding-managed-and-unmanaged-attributes)?

If so, then try setting `Realm Settings > User profile > Edit attribute > Permission > Who can view: User` as [shown here](https://www.keycloak.org/docs/latest/server_admin/images/user-profile-select-options-custom-validator.png).

You might also want to consider creating an [attribute group](https://www.keycloak.org/docs/latest/server_admin/#managing-attribute-groups) if you haven't already.

**Comment by lucamontano98** [comment]
Thank you a lot, I tried both solutions but the result is the same. How can I add new claims into "claims_supported" in the OIDC Provider metadata? Thanks. I saw that in the OIDC Core metadata of the Provider (http://localhost:8080/realms/master/.well-known/openid-configuration) there are default claims which a client can potentially ask. How can I add a new claim to this collection? Thanks.

Best regards.

Luca Montano

**Reply by codespearhead** [reply]
The `claims_supported` field is supposed to be dynamically-generated based on the claims available through all the defined scopes and their associated mappers.

Try doing as the comment in [this answer](https://stackoverflow.com/a/63937331) says:

> In newer keycloak versions (right now 20) the click path is: client -> (pick yours) -> client scopes -> pick the first (dedicated client scope) -> add mappers

Question possibly related to #10905 , so I suggest upvoting it to help that helper feature making it into the main branch.
